languages/python/reference/exercise-concepts/phone-number.md

```diff
@@ -0,0 +1,75 @@
+[Example Solution](https://github.com/exercism/python/blob/master/exercises/phone-number/example.py)
+
+```
+import re
+
+
+class PhoneNumber:
+    def __init__(self, number):
+        self.number = self._clean(number)
+        self.area_code = self.number[:3]
+        self.exchange_code = self.number[3:6]
+        self.subscriber_number = self.number[-4:]
+
+    def pretty(self):
+        return "({}) {}-{}".format(
+            self.area_code, self.exchange_code, self.subscriber_number
+        )
+
+    def _clean(self, number):
+        return self._normalize(re.sub(r"[^\d]", "", number))
+
+    def _normalize(self, number):
+        if len(number) == 10 or len(number) == 11 and number.startswith("1"):
+            valid = number[-10] in "23456789" and number[-7] in "23456789"
+        else:
+            valid = False
+
+        if valid:
+            return number[-10:]
+        else:
+            raise ValueError("{} is not a valid phone number".format(number))
+```
+
+## Concepts
+
+**Classes**
+ - classes are defined with the `class <ClassName>:` syntax
+ - classes require an `__init__` method, whose first argument is `self`, because the result of `__init__` is a class *instance*
+ - classes can have instance *methods* which are called from an instance of the class (as opposed to class methods, called from the Class itself)
```

Comment:
Again it might be adding too much detail, but there's also static methods.

Reply:
The classes concept is going to end up huge (in a good way)!

 - classes vs class instances
 - class methods, instance methods, static methods
 - `self`
 - overloading, `super()` and inheritance
... so much more

So the more detail we think of now, the better we'll be able to logically split it up.

I'll see if I can flesh this out a bit more.